config-example.yaml

```diff
@@ -200,10 +200,20 @@ log:
   format: text
   level: info
 
-# Path to a file containg ACL policies.
-# ACLs can be defined as YAML or HUJSON.
-# https://tailscale.com/kb/1018/acls/
-acl_policy_path: ""
+## ACL
+acl:
+  # The supported policy modes are "db" and "file".
+  # If "db" is selected, the policies will be stored in the database.
+  # You will be able to get and set the ACL via the ACL APIs.
+  #
+  # If "file" is selected, the policies will be loaded from a file
+  # located at `policy_path`. It will not be stored in the databse
+  # and you will not be able to get or set the ACL via the ACL APIs.
+  policy_mode: "db"
```

Comment:
I'm open for the example config to be `db`, but the default should still be file.

Reply:
> I'm open for the example config to be db, but the default should still be file.

Personally I strongly **disagree**.
 
1. Many people still use docker.  (I know I know, it's not officially supported... still!) 
Manipulating files inside it, especially if the container is **not** running is a very hard task.

2. IMHO it is more **secure** to store ACL data inside a `db`. A separate file can be easier hacked / modified / viewed.
 After all, ACL is the "main security" config of HeadScale, telling who has access to what !
 Please keep in mind: The main reason for using VPNs is security. 
 Offering an easy-to-view / edit file to the open is against it.
 
3. Eventually more people will use this feature using a web-UI, so it would make much easier to setup with that already.

4. For those who like to do things manually (by typing text files, etc) -> changing the config to `file` will not be a big task to do, but for those who are not used to do tasks like this: it would be an "impossible nightmare".

Anyway: is the database password protected by default, so only HS can access it?
- or can other programs access it too?